packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -43,29 +46,49 @@ const slots = defineSlots<{
 const { t } = injectI18n();
 
 /**
- * Currently (visually) active option.
+ * Currently (visually) active/focused option.
  */
-const activeOption = ref<TValue>();
+const activeOption = ref<TOption | undefined>();
 
 /**
- * Sync the active option with the selected option.
+ * Sync the active option with the selected option on single select.
  */
 watch(
   () => props.modelValue,
   (newValue) => {
-    activeOption.value = newValue;
+    if (!props.multiple) {
+      activeOption.value = newValue as TOption | undefined;
+    }
   },
 );
 
 const {
   elements: { listbox, option: headlessOption, group: headlessGroup },
 } = createListbox({
   label: computed(() => props.label),
+  multiple: computed(() => props.multiple),
   selectedOption: computed(() => props.modelValue),
   activeOption,
-  onSelect: (id) => {
-    if (props.modelValue === id) emit("update:modelValue", undefined);
-    else emit("update:modelValue", id as TValue);
+  onSelect: (selectedOption) => {
+    if (props.modelValue === selectedOption) emit("update:modelValue", undefined);
+    else {
+      if (!props.multiple) {
+        emit("update:modelValue", selectedOption as ListboxModelValue<TOption, TMultiple>);
+      } else {
+        const newValues =
+          !props.modelValue || (Array.isArray(props.modelValue) && !props.modelValue.length)
+            ? []
+            : (props.modelValue as TOption[]);
+
+        const foundIndex = newValues.findIndex((option) => option === selectedOption);
+        if (foundIndex < 0) newValues.push(selectedOption);
+        else {
+          newValues.splice(foundIndex, 1);
+        }
+        emit("update:modelValue", newValues as ListboxModelValue<TOption, TMultiple>);
+        activeOption.value = selectedOption;
+      }
+    }
   },
```

Comment:
```suggestion
  onSelect: (selectedOption) => {
    if (!props.multiple) {
      const newValue = selectedOption === props.modelValue ? undefined : selectedOption;
      emit("update:modelValue", newValue as typeof props.modelValue);
      return;
    }

    const arrayValues: TOption[] = Array.isArray(props.modelValue) ? props.modelValue : [];

    const newValues = arrayValues.includes(selectedOption)
      ? arrayValues.filter((i) => i !== selectedOption)
      : [...arrayValues, selectedOption];

    emit("update:modelValue", newValues as typeof props.modelValue);
  },
```

That should do the trick

Reply:
thank youuu!!!!

Reply:
still with typecast but way prettier ^^